onnxscript/ir/passes/common/_c_api_utils.py

```diff
@@ -0,0 +1,83 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+"""Utilities for interfacing with onnx C APIs."""
+
+from __future__ import annotations
+
+import logging
+from typing import TYPE_CHECKING, Callable
+
+from onnxscript import ir
+
+if TYPE_CHECKING:
+    import onnx
+
+
+logger = logging.getLogger(__name__)
+# Temporarily remove initializers larger than this size to keep model size down
+# for the onnx.shape_inference call because it needs to serialize the model
+_BIG_TENSOR_SIZE_LIMIT = 1000  # 1KB
+
+
+def call_onnx_api(
```

Comment:
What are other usages? It looks like it's very specifically designed for shape type inference pass regarding the initializer processings?

Reply:
It is used in the checker as well. It is designed to remove the initializers temporarily to that calls to onnx capis are not constrained by the 2gb limit.

Reply:
Another possibility is to provide two primitives, one for the "before" part (removing large initializers from model) and one for the "after" part (adding the large initializers back).

Reply:
I went with the recommendation below. I wanted to ensure the model is unchanged if the call fails, which is handled by the try finally block inside the call_api function.